components/in_memory_engine/src/region_stats.rs

```diff
@@ -457,6 +411,57 @@ impl RegionStatsManager {
     }
 }
 
+struct CachedRegionStat {
+    region: Region,
+    stat: RegionStat,
+    sma_cop_requests_count: usize,
+    sma_cop_requests_avg: usize,
+    iterated_count: usize,
+}
+
+/// Sorter for `RegionStat` for cached regions.
+///
+/// Prioritizes cop requests over iterated count, assuming consistent workload
+/// patterns after caching, implying cached regions have configured MVCC
+/// amplification at least.
+fn sort_cached_region_stats(
+    region_stats: Vec<(Region, RegionStat)>,
+    cached_regions: &HashMap<u64, Arc<StdMutex<CopRequestsSMA>>>,
+) -> Vec<CachedRegionStat> {
+    // update the moving average of the coprocessor requests.
+    let mut crss = Vec::with_capacity(region_stats.len());
+    for (region, stat) in region_stats {
+        let mut cop_requests_avg = stat.query_stats.coprocessor as usize;
+        let mut cop_requests_count = 0;
+        let cop_requests = stat.query_stats.coprocessor as usize;
+        if let Some(cached_region) = cached_regions.get(&region.id)
+            && let Ok(mut avg) = cached_region.lock()
+        {
+            avg.observe(cop_requests as f64);
+            cop_requests_avg = avg.get_avg().ceil() as usize;
+            cop_requests_count = avg.get_count();
+        }
+        crss.push(CachedRegionStat {
+            iterated_count: stat.cop_detail.iterated_count(),
+            region,
+            stat,
+            sma_cop_requests_avg: cop_requests_avg,
+            sma_cop_requests_count: cop_requests_count,
+        });
+    }
+
+    crss.sort_by(|a, b| {
```

Comment:
Ideally, we should cache regions which to most `reqeusts * (avg_kv_per_req_before_cache - avg_kv_per_req_after_cache)`, this can largely reflect the resource usage save after load. Thus, I think QPS or request_count is an even poorer metrics here.

Reply:
We can iterative improve the eviction algorithm, after all the master implementation is inaccurate, and this PR can reduce spike as tests show.